core/translate/optimizer.rs

```diff
@@ -293,19 +295,23 @@ impl Optimizable for ast::Expr {
         &mut self,
         table_index: usize,
         table_reference: &TableReference,
-        available_indexes: &[Rc<Index>],
-    ) -> Result<Option<usize>> {
+        available_indexes: &HashMap<String, Vec<Rc<Index>>>,
+    ) -> Result<Option<Rc<Index>>> {
         match self {
             Self::Column { table, column, .. } => {
                 if *table != table_index {
                     return Ok(None);
                 }
-                for (idx, index) in available_indexes.iter().enumerate() {
-                    if index.table_name == table_reference.table.get_name() {
-                        let column = table_reference.table.get_column_at(*column);
-                        if index.columns.first().unwrap().name == column.name {
-                            return Ok(Some(idx));
-                        }
+                let available_indexes_for_table =
+                    available_indexes.get(table_reference.table.get_name());
+                if available_indexes_for_table.is_none() {
+                    return Ok(None);
+                }
+                let available_indexes_for_table = available_indexes_for_table.unwrap();
+                for index in available_indexes_for_table.iter() {
+                    let column = table_reference.table.get_column_at(*column);
```

Comment:
This can be extracted outside of the loop, right?

Reply:
yes